docs/usage/shoots.md

```diff
@@ -17,3 +17,27 @@ In order to delete your cluster, you have to set an annotation confirming the de
 ```bash
 $ ./hack/delete-shoot johndoe-1
 ```
+
+# Updating Shoot Cluster version and How Auto Update Feature is Handled
+
+If a shoot has `.spec.maintenance.autoUpdate.kubernetesVersion: true` in the manifest, and you update the `.spec.<provider>.constraints.kubernetes.versions` field in the CloudProfile used in the Shoot, then Gardener will apply Kubernetes [patch releases](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#patch-releases) updates automatically during the `.spec.maintenance.timeWindow`.
+
+Since Kubernetes follows [Semantic Versioning](http://semver.org/), if indicated so, Gardener will automatically apply the patch release updates. But it will never auto update the Major or Minor releases since there is no effort to keep backward compatibility in those releases. Major or Minor updates must be handled by updating the `.spec.kubernetes.version` field manually. Before applying such update on Minor or Major releases, operators should check for all the breaking chances introduced in the target release Changelog.
```

Comment:
Can we also mention here that major and minor updates, will immediately executed after changing ``spec.kubernetes.version`` and will not wait until the update time window reached?

I would also recommend to highlight the last sentence, because this is an important statement and if not considered, it could lead in the worst case to workload incompatibilities and outages.